apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js

```diff
@@ -332,26 +332,35 @@ export function fetchStandardsCoveredForScript(scriptId) {
 export function fetchStudentLevelScores(scriptId, sectionId) {
   return (dispatch, getState) => {
     let state = getState();
-    $.ajax({
-      method: 'GET',
-      dataType: 'json',
-      url: `/dashboardapi/v1/teacher_scores/${sectionId}/${scriptId}`
-    }).then(data => {
-      const scoresData = data;
-      dispatch(setStudentLevelScores(scoresData));
-      let initialCompletedUnpluggedLessons = getInitialUnpluggedLessonCompletionStatus(
-        state,
-        scriptId,
-        scoresData
-      );
-      let unpluggedLessonList = getUnpluggedLessonsForScript(state);
-      const lessonsToSelect = _.filter(unpluggedLessonList, function(lesson) {
-        if (initialCompletedUnpluggedLessons.includes(lesson.id)) {
-          return lesson;
-        }
-      });
-      dispatch(setSelectedLessons(lessonsToSelect));
+    const numStudents =
+      state.teacherSections.sections[state.teacherSections.selectedSectionId]
+        .studentCount;
+    const NUM_STUDENTS_PER_PAGE = 50;
+    const numPages = Math.ceil(numStudents / NUM_STUDENTS_PER_PAGE);
+    const requests = _.range(1, numPages + 1).map(currentPage => {
+      const url = `/dashboardapi/v1/teacher_scores/${sectionId}/${scriptId}?page=${currentPage}`;
+      return fetch(url, {credentials: 'include'})
+        .then(response => response.json())
+        .then(data => {
+          const scoresData = data;
+          dispatch(setStudentLevelScores(scoresData));
```

Comment:
I think right now the last request to finish "wins" here and clobbers `studentLevelScoresByStage` in the redux state.  We probably need to update this part of the reducer function to merge in each page of the scores data as it arrives.

https://github.com/code-dot-org/code-dot-org/blob/8741a3a13239bef056836543b06f00eecba43557/apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js#L67-L72

Reply:
I think everything after this line should move to happen after the `Promise.all` so we are making the `setSelectedLessons` decision after we've downloaded all of the level scores data.

Reply:
> I think right now the last request to finish "wins" here and clobbers `studentLevelScoresByStage` in the redux state. We probably need to update this part of the reducer function to merge in each page of the scores data as it arrives.
> 
> https://github.com/code-dot-org/code-dot-org/blob/8741a3a13239bef056836543b06f00eecba43557/apps/src/templates/sectionProgress/standards/sectionStandardsProgressRedux.js#L67-L72

You got it - before I was overwriting any teacher score data already in the state so we ended up with only 50 students' scores per stage at the end. 
![overwriting-teacher-scores](https://user-images.githubusercontent.com/12300669/78322161-c3a61200-7522-11ea-96ae-49b254d546f6.gif)

With the changes in 7911510, I add the newly fetched batch of teacher scores to the already accumulated teacher scores resulting in the expected scores for 250 students. 
![adding-teacher-scores](https://user-images.githubusercontent.com/12300669/78322225-fb14be80-7522-11ea-9bc2-784529407add.gif)